# Prop. 67: Plastic Bag Ban
After [First Python Notbook](http://www.firstpythonnotebook.org "First Python Notbook") by **Ben Welsh**. Comments (partly in in Swedish) by <a href="mailto:sanna.volny@gmail.com?Subject=First_Python_Notebook" target="_top">Sanna Volny</a>.

In [1]:
import pandas as pd
# importera csv-filer till variblerna props (kommittéer som stödjer/motsätter sig propositioner i valet 2016 i Kalifornien) och contribs (kampanjbidragsgivare)
props = pd.read_csv("http://www.firstpythonnotebook.org/_static/committees.csv")
contribs = pd.read_csv("http://www.firstpythonnotebook.org/_static/contributions.csv")
# skapa en variabel av alla kommittéer som stödjer/motsätter sig proposition 67 om att förbjuda plastpåsar
prop67 = props[props.prop_name == 'PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON SINGLE-USE PLASTIC BAGS.']
# Visa de 5 första posterna
prop67.head()

ocd_prop_id  calaccess_prop_id  \
95  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
96  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
97  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
98  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
99  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

    ccdc_prop_id                                          prop_name  \
95            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
96            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
97            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
98            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
99            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

    ccdc_committee_id  calaccess_committee_id  \
95                477                 1372900   
96                478                 1372902   
97                479                 1329332   
98                480                 1374885   
99                481                 1346973   

                                       committee_name committee_position  
95  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT  
96  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE  
97  SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...            SUPPORT  
98  YES ON 67 - CALIFORNIANS AGAINST WASTE - PROTE...            SUPPORT  
99  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...            SUPPORT

** How many supports/opposes the plastic bag ban in California? **

In [2]:
merged = pd.merge(prop67, contribs, on="calaccess_committee_id")
merged.committee_position.value_counts()

SUPPORT    6252
OPPOSE       27
Name: committee_position, dtype: int64

** How much money does each side get and what is the avarage contribution on each side? **

In [3]:
# OPPOSE:
merged[merged.committee_position == 'OPPOSE'].amount.sum()

2681413.25

In [4]:
#merged[merged.committee_position == 'OPPOSE'].calaccess_committee_id.count()
merged[merged.committee_position == 'OPPOSE'].amount.sum() / merged[merged.committee_position == 'OPPOSE'].calaccess_committee_id.count()

99311.601851851854

In [5]:
# SUPPORT:
merged[merged.committee_position == 'SUPPORT'].amount.sum()

1789160.75

In [6]:
#merged[merged.committee_position == 'SUPPORT'].calaccess_committee_id.count()
merged[merged.committee_position == 'SUPPORT'].amount.sum() / merged[merged.committee_position == 'SUPPORT'].calaccess_committee_id.count()

286.17414427383238

**How much more does the average opposer give, compared to the avarage supporter?**

In [7]:
(merged[merged.committee_position == 'OPPOSE'].amount.sum() / merged[merged.committee_position == 'OPPOSE'].calaccess_committee_id.count())/(merged[merged.committee_position == 'SUPPORT'].amount.sum() / merged[merged.committee_position == 'SUPPORT'].calaccess_committee_id.count())

347.03205666554993

So the average opposer gives 347 times as much! My hypothesis that the environmentalists give a little money per person and that those who oppose the plastic ban represents commercial interests and give more money seem to be right. Or? Who are the committées and contributers?

In [8]:
merged.groupby(["committee_name_x", "committee_position"]).amount.sum().reset_index().sort_values('amount', ascending=False)

committee_name_x committee_position  \
0  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
5  YES ON 67 - PROTECT THE PLASTIC BAG BAN, SPONS...            SUPPORT   
2  SAVE THE BAG BAN, YES ON 67, SPONSORED BY ENVI...            SUPPORT   
1  CALIFORNIA GROCERS ASSOCIATION ISSUES COMMITTE...            SUPPORT   
3  SAVE THE BAY ACTION FUND COMMITTEE TO SUPPORT ...            SUPPORT   
4  YES ON 67 - CALIFORNIANS AGAINST WASTE - PROTE...            SUPPORT   

       amount  
0  2681413.25  
5   861295.75  
2   609365.00  
1   200000.00  
3    97050.00  
4    21450.00

In [22]:
# The biggest contributors:
merged.groupby(["contributor_firstname", "contributor_lastname", "committee_position"]).amount.sum().reset_index().sort_values("amount", ascending=False).head()

contributor_firstname contributor_lastname committee_position    amount
1019                 JULIE              PACKARD            SUPPORT  105000.0
330                 CLAIRE                PERRY            SUPPORT  100000.0
1912              THOMAS F               STEYER            SUPPORT   50000.0
102                ANNE G.              EARHART            SUPPORT   25000.0
1466                 NANCY              BURNETT            SUPPORT   19000.0

Nope. The biggest contributers seams to support the ban. But can that really be true? Lets look at those who oppose.

In [23]:
oppose = merged[merged.committee_position == 'OPPOSE']
oppose.sort_values("amount", ascending=False).head()

ocd_prop_id  calaccess_prop_id  \
141  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
132  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
140  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
142  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   
133  ocd-contest/7495cdbe-1aa7-4c26-9a55-aa4130347b95            1372638   

     ccdc_prop_id                                          prop_name  \
141            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
132            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
140            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
142            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   
133            86  PROPOSITION 067- REFERENDUM TO OVERTURN BAN ON...   

     ccdc_committee_id  calaccess_committee_id  \
141                478                 1372902   
132                478                 1372902   
140                478                 1372902   
142                478                 1372902   
133                478                 1372902   

                                      committee_name_x committee_position  \
141  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
132  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
140  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
142  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   
133  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...             OPPOSE   

                                      committee_name_y  calaccess_filing_id  \
141  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
132  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   
140  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
142  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2038213   
133  AMERICAN PROGRESSIVE BAG ALLIANCE, A PROJECT O...              2012817   

    date_received                 contributor_lastname contributor_firstname  \
141    2016-01-25  FORMOSA PLASTICS CORPORATION U.S.A.                   NaN   
132    2015-10-30  FORMOSA PLASTICS CORPORATION U.S.A.                   NaN   
140    2016-01-13                   HILEX POLY CO. LLC                   NaN   
142    2016-01-27                   HILEX POLY CO. LLC                   NaN   
133    2015-11-17                   HILEX POLY CO. LLC                   NaN   

    contributor_city contributor_state contributor_zip contributor_employer  \
141       LIVINGSTON                NJ            7039                  NaN   
132       LIVINGSTON                NJ            7039                  NaN   
140       HARTSVILLE                SC           29550                  NaN   
142       HARTSVILLE                SC           29550                  NaN   
133       HARTSVILLE                SC           29550                  NaN   

    contributor_occupation  contributor_is_self_employed    amount  
141                    NaN                         False  356441.5  
132                    NaN                         False  350000.0  
140                    NaN                         False  305675.0  
142                    NaN                         False  299175.0  
133                    NaN                         False  186585.0

Hm. The biggest contributers among those who oppose the ban give three times as much as the biggest supporters. So why didn´t I catch them in my last inquiry? It must be because its companies and that the values are missing in the field firstname.

Let´s try again but only with the contributers last name.

In [21]:
merged.groupby(["contributor_lastname", "committee_position"]).amount.sum().reset_index().sort_values("amount", ascending=False).head()

contributor_lastname committee_position     amount
713                    HILEX POLY CO. LLC             OPPOSE  874769.00
522   FORMOSA PLASTICS CORPORATION U.S.A.             OPPOSE  748441.50
1534                       SUPERBAG CORP.             OPPOSE  609369.75
12                  ADVANCE POLYBAG, INC.             OPPOSE  446833.00
256        CALIFORNIA GROCERS ASSOCIATION            SUPPORT  200000.00

That was more like what I have expected.Tricky! 